File: MultiplesOfANumber/moan.js
// Multiples of a Number soln in Node for CodeEval.com by Steven A. Dunn
var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function(line) {
  if (line !== "") {
    line = line.split(",");
    var x = parseInt(line[0]);
    var n = parseInt(line[1]);

    var multiple = 2;
    var multipleN = n;

    while (x > multipleN) {
      multipleN = n * multiple;
      multiple += 1;
    }

    console.log(multipleN);
  }
});

File: MultiplesOfANumber/c/moan.c
// multiples of a number soln in c for code eval by steven a dunn

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char* argv[])
{
    FILE *fp = fopen(argv[1], "r");
    if (fp != NULL)
    {
        char line[1024];
        while (fgets(line, sizeof(line), fp))
        {
            int x = 0;
            int n = 0;
            char *token = strtok(line, ",");
            x = atoi(token);
            while (token != NULL)
            {
                token = strtok(NULL, ",");
                if (token != NULL)
                    n = atoi(token);
            }

            int multiple = 2;
            int multipleN = n;
            while (x > multipleN)
            {
                multipleN = n * multiple;
                multiple += 1;
            }
            printf("%d\n", multipleN);
        }
        fclose(fp);
    }
    return 0;
}